### PR TITLE
Correct sampled and debug B3 headers

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
Distinguishes between X-B3-Sampled and X-B3-Flags B3 headers on
context injection and adds case insensitivity for carrier key
checks.

Resolves: https://github.com/jaegertracing/jaeger-client-python/issues/120

Signed-off-by: Ryan Fitzpatrick <rmfitzpatrick@signalfx.com>

### CODE REVIEW

File: jaeger_client/codecs.py [hunk 1]
```diff
@@ -271,21 +271,25 @@ def inject(self, span_context, carrier):
...
+        elif span_context.flags & SAMPLED_FLAG == SAMPLED_FLAG:
+            carrier[self.sampled_header] = '1'
 
     def extract(self, carrier):
         if not isinstance(carrier, dict):
             raise InvalidCarrierException('carrier not a dictionary')
-        trace_id = header_to_hex(carrier.get(self.trace_header.lower()))
-        span_id = header_to_hex(carrier.get(self.span_header.lower()))
-        parent_id = carrier.get(self.parent_span_header.lower(), None)
+        ic_map = dict([(k.lower(), k) for k in carrier])
```
Comment (yurishkuro) [inline jaeger_client/codecs.py]: what does `ic` stand for?

File: jaeger_client/codecs.py [hunk 2]
```diff
@@ -271,21 +271,25 @@ def inject(self, span_context, carrier):
...
+        trace_id = header_to_hex(carrier.get(ic_map.get(self.trace_header.lower())))
+        span_id = header_to_hex(carrier.get(ic_map.get(self.span_header.lower())))
+        parent_id = carrier.get(ic_map.get(self.parent_span_header.lower()))
```
Comment (yurishkuro) [inline jaeger_client/codecs.py]: Calling `.lower()` on a constant on every request seems highly inefficient.  Why not cache these on construction?

File: jaeger_client/codecs.py (hunk 1 above)
Comment (rmfitzpatrick) [reply jaeger_client/codecs.py]: "Ignore case".  Updated to `lowercase_keys` for clarity.

File: jaeger_client/codecs.py (hunk 2 above)
Comment (rmfitzpatrick) [reply jaeger_client/codecs.py]: These changes don't introduce this practice, but good idea.  Updated to use class attributes for single initialization.

**Pull request merged**